File: src/types/company.ts
import { Booking } from './booking';
import { Employee } from './employee';

export interface Company {
    id: number;
    name: string;
    address?: string;
    city?: string;
    state?: string;
    postalCode?: string;
    country?: string;
    phone?: string;
    email?: string;
    taxId?: string;
    isActive: boolean;
    createdAt: string;
    updatedAt?: string;
    bookings?: Booking[];
    employees?: Employee[];
}

export interface CompanyCreate {
    name: string;
    address?: string;
    city?: string;
    state?: string;
    postalCode?: string;
    country?: string;
    phone?: string;
    email?: string;
    taxId?: string;
}

export type CompanyUpdate = Partial<CompanyCreate> & {
    isActive?: boolean;
}; 

File: Front/SmartStayHaven.API/Models/Company.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SmartStayHaven.API.Models
{
    public class Company
    {
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;

        [StringLength(200)]
        public string Address { get; set; } = string.Empty;

        [StringLength(50)]
        public string City { get; set; } = string.Empty;

        [StringLength(50)]
        public string State { get; set; } = string.Empty;

        [StringLength(20)]
        public string PostalCode { get; set; } = string.Empty;

        [StringLength(50)]
        public string Country { get; set; } = string.Empty;

        [StringLength(20)]
        public string Phone { get; set; } = string.Empty;

        [StringLength(100)]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;

        [StringLength(50)]
        public string TaxId { get; set; } = string.Empty;

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public virtual ICollection<Booking> Bookings { get; set; } = new List<Booking>();
        public virtual ICollection<Employee> Employees { get; set; } = new List<Employee>();
    }
} 